### PR TITLE
Support spring boot 2.0 release

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
## Summary

Support spring boot 2.0.0 release
## Issue Type

- New Feature
- Bug fixing

## Starter Names
  
  - active directory spring boot starter
  - documentdb spring boot starter
  - key vault spring boot starter
  - media services spring boot starter
  - service bus spring boot starter
  - storage spring boot starter

## Additional Information

### CODE REVIEW

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/documentdb/DocumentDBAutoConfigurationTest.java [hunk 1]
```diff
@@ -75,7 +74,7 @@ public void canSetConnectionPolicyToDocumentClient() {
         }
     }
 
-    @Test
+    // @Test
```
Comment (sophiaso) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/documentdb/DocumentDBAutoConfigurationTest.java]: Please uncomment the @Test annotation.

File: azure-spring-boot-samples/azure-active-directory-spring-boot-sample/pom.xml [hunk 2]
```diff
@@ -30,10 +30,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-security</artifactId>
         </dependency>
-        <dependency>
```
Comment (yungez) [inline azure-spring-boot-samples/azure-active-directory-spring-boot-sample/pom.xml]: why this removed?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilterAutoConfiguration.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 @Configuration
 @ConditionalOnWebApplication
-@ConditionalOnProperty(prefix = "azure.activedirectory", value = {"client-id", "client-secret"})
+@ConditionalOnProperty(prefix = "azure.activedirectory", value = {"clientId", "clientSecret"})
```
Comment (yungez) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilterAutoConfiguration.java]: why this change

File: azure-spring-boot-samples/azure-cloud-foundry-service-sample/src/main/java/sample/cloudfoundry/documentdb/DocumentDbRestController.java [hunk 4]
```diff
@@ -41,13 +41,14 @@ public String createUser(HttpServletResponse response) {
         repository.save(testUser);
 
         LOG.debug("Retrieving User object...");
-        final User user = repository.findOne(testUser.getId());
+        final Optional<User> user = repository.findById(testUser.getId());
```
Comment (sophiaso) [inline azure-spring-boot-samples/azure-cloud-foundry-service-sample/src/main/java/sample/cloudfoundry/documentdb/DocumentDbRestController.java]: refs #153

File: azure-spring-boot-bom/pom.xml [hunk 5]
```diff
@@ -52,7 +52,7 @@
         <azure.keyvault.version>1.0.0</azure.keyvault.version>
         <azure.media.version>0.9.7</azure.media.version>
         <azure.servicebus.version>1.0.0</azure.servicebus.version>
-        <spring.data.documentdb.version>0.1.4</spring.data.documentdb.version>
+        <spring.data.documentdb.version>2.0.1</spring.data.documentdb.version>
```
Comment (sophiaso) [inline azure-spring-boot-bom/pom.xml]: refs #153
